helpers/html_converter.sh

```diff
@@ -0,0 +1,248 @@
+#!/bin/bash
+
+# emba - EMBEDDED LINUX ANALYZER
+#
+# Copyright 2020-2021 Siemens AG
+#
+# emba comes with ABSOLUTELY NO WARRANTY. This is free software, and you are
+# welcome to redistribute it under the terms of the GNU General Public License.
+# See LICENSE file for usage of this software.
+#
+# emba is licensed under GPLv3
+#
+# Author(s): Stefan Haboeck
+
+declare -a MENU_LIST
+
+build_index_file(){
+  
+  local TOP10_FORMAT_COUNTER
+  local HTML_FILE
+  FILE=$1
+  FILENAME=$(basename "$FILE")
+  HTML_FILE="$(basename "${FILE%.txt}"".html")"
+  COULORLESS_FILE_LINE=$(head -1 "$FILE" | tail -n 1 | cut -c27-)
+  
+  if [[ ${FILENAME%.txt} == "s05"* ]] || [[ ${FILENAME%.txt} == "s25"* ]]; then
+    if [[ "$(wc -l "$FILE" | cut -d\  -f1 2>/dev/null)" -gt 0 ]] ;  then
+      readarray -t STRING_LIST <"$FILE"
+      INDEX_CONTENT_ARR+=("${STRING_LIST[@]}")
+    fi
+  elif [[ ${FILENAME%.txt} == "f50"* ]]; then
+    if [[ "$(wc -l "$FILE" | cut -d\  -f1 2>/dev/null)" -gt 0 ]] ;  then
+      readarray -t STRING_LIST <"$FILE"
+      INDEX_CONTENT_ARR=("${STRING_LIST[@]}")
+    fi
+  fi
+ 
+  MENU_LIST+=("<li><a href=\"$HTML_FILE\">$COULORLESS_FILE_LINE</a></li>")
+
+  if [ ${#FILENAMES[@]} -ne 0 ]; then
+    FILENAMES[${#FILENAMES[@]}]="${FILENAME%.txt}"
+  else
+    FILENAMES[0]="${FILENAME%.txt}"
+  fi
+
+  echo "$HTML_FILE_HEADER<div><ul>" | tee -a "$HTML_PATH""/index.txt" >/dev/null
+      	
+  if [[ -n ${MENU_LIST[*]} ]]; then
+    for OUTPUT in "${MENU_LIST[@]}"; do
+      echo -e "$OUTPUT" | tee -a "$HTML_PATH""/index.txt" >/dev/null	
+    done
+  fi
+  
+  if test -f "$HTML_PATH""/collection.html"; then
+    echo "<li><a href=""$HTML_PATH""/collection.html"">""Nothing found""</a></li>" | tee -a "$HTML_PATH""/index.txt" >/dev/null
+  fi
+  
+  echo "</ul></div><div class=\"main\">"| tee -a "$HTML_PATH""/index.txt" >/dev/null
+  if [[ ${FILENAME%.txt} != "f"* ]]; then
+    echo "<h2>[[0;34m+[0m] [0;36m[1mGerneral Information[0m[1m[0m</h2>
+      File: $(basename "$FIRMWARE_PATH")<br>
+      Architecture: $ARCH<br>
+      Date: $(date) <br>
+      Duration time: $(date -d@$SECONDS -u +%H:%M:%S) <br>
+      EMBA Command: $EMBACOMMAND <br>
+      " | tee -a "$HTML_PATH""/index.txt" >/dev/null
+  fi
+     	
+  i=0
+  TOP10_FORMAT_COUNTER=0
+  if [[ -n ${INDEX_CONTENT_ARR[*]} ]]; then
+    for OUTPUT in "${INDEX_CONTENT_ARR[@]}"; do
+      if [[ "$OUTPUT" == *"entropy.png"* ]]; then
+        #heigth=\"300px\" width=\"300px\"
+	OUTPUT=${OUTPUT//"33m"/"33m <br> <img id=\"entropypic\" heigth=\"380px\" width=\"540px\" src=\""}
```

Comment:
also misaligned -I will comment this only once, please check your code for right alignment.